### DISCUSSION TITLE
Typehint iterable/Iterator::current type

### DISCUSSION THREAD

**Discussion by bmewburn** [phpDocumentor/phpDocumentor#3646] [PSR]
I see there has been some work and discussion on templates which may make this suggestion obsolete but it seems to have been going on forever and is stalled.

Arrays have a shorthand syntax `int[]` what about something similar for iterables `int*`? `*` because of  it's use as 0 or many in various notations.

eg
```php
<?php

/**
 * @return int*
 */
function gen1()
{
    yield 1;
}

/**
 * @return (int|MyClass)*
 */
 function gen2()
 {
     yield 1;
     yield new MyClass;
 }
```

**Comment by mvriel** [comment]
I'm afraid I need to defer this to @ashnazg as he is busy with PSR-5. Chuck, do you have a place where you track these suggestions?
